src/pages/index.js

```diff
@@ -1,76 +1,56 @@
-	// Data
-	const {data, loading, error} = useFetch('/api/hello');
+const API_KEY = 'cfe8f1e1a9b233b64412ec3cd0525b67';
+const GET_CONFIGURATION = 'https://api.themoviedb.org/3/configuration?api_key=' + API_KEY;
+
+//const GET_MOVIES = `https://api.themoviedb.org/3/discover/movie?api_key=${API_KEY}&language=en-US&sort_by=popularity.desc&include_adult=false&include_video=false&page=1`;
 
-	// Local state
-	const [id, setId] = useState(null);
+const GET_TV = `https://api.themoviedb.org/3/discover/tv?api_key=${API_KEY}&language=en-US&sort_by=popularity.desc&include_adult=false&include_video=false&page=1`;
 
+export default function HomePage() {
 	// Global state
-	const counter = useStore(state => state.counter);
-	const decrementCounter = useStore(state => state.decrementCounter);
-	const incrementCounter = useStore(state => state.incrementCounter);
-	const setCounter = useStore(state => state.setCounter);
+	const setConfiguration = useStore(state => state.setConfiguration);
+	const setData = useStore(state => state.setData);
+	const setConfigurationLoaded = useStore(state => state.setConfigurationLoaded);
+	const isConfigurationLoaded = useStore(state => state.isConfigurationLoaded);
+
+	useEffect(() => {
+		Promise.all([fetch(GET_TV), fetch(GET_CONFIGURATION)])
+			.then(function (responses) {
+				return Promise.all(
+					responses.map(function (response) {
+						return response.json();
+					})
+				);
+			})
+			.then(function (data) {
+				console.log('Promise.all() should be done.');
+				console.log(data);
+				setConfiguration(data[1].images);
+				setData(data[0].results);
+				setConfigurationLoaded(true);
+			})
+			.catch(function (error) {
+				console.log(error);
+			});
+	}, [setConfiguration, setConfigurationLoaded, setData]);
 
 	return (
 		<Layout>
 			<Head>
-				<title key="title">My Project</title>
-				<meta key="description" name="description" content="This is my project" />
-			</Head>
-			<h1>Home</h1>
-			{loading && <div>Loading...</div>}
-			{error && <div>{error.message}</div>}
-			{data && (
-				<pre>
-					<code>{JSON.stringify(data, null, 4)}</code>
-				</pre>
-			)}
-			<section>
-				<Button
-					aria-label="decrement"
-					onClick={() => {
-						decrementCounter();
-					}}
-				>
-					-
-				</Button>
-				<input
-					value={`${counter}`}
-					size={2}
-					onChange={event => {
-						setCounter(Number.parseInt(event.target.value, 10));
-					}}
+				<title key="title">Favorite Movie App</title>
+				<meta
+					key="description"
+					name="description"
+					content="Robert Reichs capstone project"
 				/>
-				<Button
-					aria-label="increment"
-					onClick={() => {
-						incrementCounter();
-					}}
-				>
-					+
-				</Button>
-			</section>
-			<br />
-			<section>
-				<Button
-					onClick={() => {
-						setId(nanoid());
-					}}
-				>
-					Generate ID
-				</Button>
-				<br />
-				<div>Id: {id}</div>
-			</section>
+			</Head>
+			{isConfigurationLoaded ? <MovieList /> : null}
```

Comment:
If the movies cannot be loaded, there should be an error message displayed for the user.

Reply:
This is not checking if the movie is loading, its checking if the configuration is loaded. Prevents the Component to be rendered twice, once with undefined and a 2nd time when the data is loaded.

Reply:
But what happens if the configuration is not loaded? Then the movies will not be fetched I assume?

Reply:
I recommend to follow Andreas suggestions - but it's up to you and it's your code quality.  😉